File: app/src/main/java/net/rmoreno/weatherapp/presenters/SettingsPresenterImpl.java
package net.rmoreno.weatherapp.presenters;

import net.rmoreno.weatherapp.SettingsInteractor;

public class SettingsPresenterImpl implements SettingsPresenter {

    private SettingsInteractor interactor;

    public SettingsPresenterImpl(SettingsInteractor interactor) {
        this.interactor = interactor;
    }
    @Override
    public void resume() {}

    @Override
    public void destroy() {}

    @Override
    public void pause() {}

    @Override
    public void updateSweather(int temperature) {
        interactor.updateSweather(temperature);
    }
}

File: app/src/main/java/net/rmoreno/weatherapp/presenters/SettingsPresenterImpl.kt
package net.rmoreno.weatherapp.presenters

import net.rmoreno.weatherapp.SettingsInteractor

class SettingsPresenterImpl(private val interactor: SettingsInteractor) : SettingsPresenter {
    override fun resume() {}

    override fun destroy() {}

    override fun pause() {}

    override fun updateSweather(temperature: Int) {
        interactor.updateSweather(temperature)
    }
}
